### PR TITLE
add prometheus svt to pipeline

### PR DESCRIPTION
**Pull request by mrsiano**
extend the pipeline within prometheus workloads
@chaitanyaenr @akrzos ptal

### CODE REVIEW

File: alderaan/pipeline-scripts/prometheus.groovy [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		currentBuild.result = "SUCCESS"
+		node(node_label) {
+			if (fileExists(property_file_name)) {
+				println "prometheus.properties file exist... deleting it..."
+				sh "rm ${property_file_name}"
+			}
+			sh "wget -O ${property_file_name} ${PROMETHEUS_PROPERTY_FILE}"
+			sh "cat ${property_file_name}"
+			def prometheus_properties = readProperties file: property_file_name
+			def REFRESH_INTERVAL = prometheus_properties['REFRESH_INTERVAL']
```
Comment (chaitanyaenr) [inline alderaan/pipeline-scripts/prometheus.groovy]: All the parameters need by the job should be read from the properties including jump_host, user, containerized e.t.c

File: alderaan/pipeline-scripts/prometheus.groovy [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+			println "-------------------------------------------------"
+			println "-------------------------------------------------"
+
+			// copy the parameters file to jump host
+			sh "git clone https://${token}@${repo} ${WORKSPACE}/perf-dept && chmod 600 ${WORKSPACE}/perf-dept/ssh_keys/id_rsa_perf"
+			sh "scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${WORKSPACE}/perf-dept/ssh_keys/id_rsa_perf ${property_file_name} root@${jump_host}:/root/properties"
+
+			try {
+				prometheus_build = build job: 'SVT_PROMETHEUS_SCALING',
+				parameters: [	[$class: 'StringParameterValue', name: 'NAMESPACE', value: NAMESPACE ],
```
Comment (chaitanyaenr) [inline alderaan/pipeline-scripts/prometheus.groovy]: All the parameters needed by the prometheus job should be passed by this script. We shouldn't have default values in the prometheus jenkins job.

File: alderaan/property_files/prometheus.properties [hunk 3]
```diff
@@ -0,0 +1,20 @@
+JUMP_HOST=
+USER=root
+SETUP_PBENCH=false
```
Comment (chaitanyaenr) [inline alderaan/property_files/prometheus.properties]: SETUP_PBENCH, ENVIRONMENT, TOOLING_INVENTORY are not needed. We will handle the pbench setup using a separate job ( SETUP-TOOLING ) which is going to run just once after openshift install and we will use the same tools for all the tests.

File: alderaan/property_files/prometheus.properties [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+USE_PROXY=false
+CONTAINERIZED=false
+PROXY_USER=""
+PROXY_HOST=""
+ENVIRONMENT=alderaan
+GITHUB_TOKEN=
+PERF_REPO=
+PBENCH_SERVER=
+SCALE_CI_RESULTS_TOKEN=
+JOB=nodevertical
```
Comment (chaitanyaenr) [inline alderaan/property_files/prometheus.properties]: s/nodevertical/prometheus

File: alderaan/pipeline-scripts/prometheus.groovy [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+			try {
+				prometheus_build = build job: '${JOB}',
+				parameters: [
+						[$class: 'StringParameterValue', name: 'JUMP_HOST', value: JUMP_HOST ],
+						[$class: 'StringParameterValue', name: 'USER', value: USER ],
+						[$class: 'StringParameterValue', name: 'CLEAR_RESULTS', value: CLEAR_RESULTS ],
+						[$class: 'StringParameterValue', name: 'MOVE_RESULTS', value: MOVE_RESULTS ],
+						[$class: 'StringParameterValue', name: 'USE_PROXY', value: USE_PROXY ],
+						[$class: 'StringParameterValue', name: 'CONTAINERIZED', value: CONTAINERIZED ]]
```
Comment (chaitanyaenr) [inline alderaan/pipeline-scripts/prometheus.groovy]: The parameters are comma separated values.

File: alderaan/pipeline-scripts/prometheus.groovy [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+						[$class: 'StringParameterValue', name: 'PROXY_HOST', value: PROXY_HOST ],
+						[$class: 'StringParameterValue', name: 'GITHUB_TOKEN', value: GITHUB_TOKEN ]]
+						[$class: 'StringParameterValue', name: 'PERF_REPO', value: PERF_REPO ],
+						[$class: 'StringParameterValue', name: 'PBENCH_SERVER', value: PBENCH_SERVER ],
+						[$class: 'StringParameterValue', name: 'SCALE_CI_RESULTS_TOKEN', value: SCALE_CI_RESULTS_TOKEN ],
+						[$class: 'StringParameterValue', name: 'JOB', value: JOB ]]
+						[$class: 'StringParameterValue', name: 'REFRESH_INTERVAL', value: REFRESH_INTERVAL ]]
+						[$class: 'StringParameterValue', name: 'CONCURRENCY', value: CONCURRENCY ]]
+						[$class: 'StringParameterValue', name: 'GRAPH_PERIOD', value: GRAPH_PERIOD ]]
+						[$class: 'StringParameterValue', name: 'DURATION', value: DURATION ]]
```
Comment (chaitanyaenr) [inline alderaan/pipeline-scripts/prometheus.groovy]: Ditto

File: alderaan/pipeline-scripts/prometheus.groovy [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+						[$class: 'StringParameterValue', name: 'PROXY_USER', value: PROXY_USER ],
+						[$class: 'StringParameterValue', name: 'PROXY_HOST', value: PROXY_HOST ],
+						[$class: 'StringParameterValue', name: 'GITHUB_TOKEN', value: GITHUB_TOKEN ]]
+						[$class: 'StringParameterValue', name: 'PERF_REPO', value: PERF_REPO ],
+						[$class: 'StringParameterValue', name: 'PBENCH_SERVER', value: PBENCH_SERVER ],
+						[$class: 'StringParameterValue', name: 'SCALE_CI_RESULTS_TOKEN', value: SCALE_CI_RESULTS_TOKEN ],
+						[$class: 'StringParameterValue', name: 'JOB', value: JOB ]]
+						[$class: 'StringParameterValue', name: 'REFRESH_INTERVAL', value: REFRESH_INTERVAL ]]
+						[$class: 'StringParameterValue', name: 'CONCURRENCY', value: CONCURRENCY ]]
+						[$class: 'StringParameterValue', name: 'GRAPH_PERIOD', value: GRAPH_PERIOD ]]
```
Comment (chaitanyaenr) [inline alderaan/pipeline-scripts/prometheus.groovy]: Ditto

File: alderaan/pipeline-scripts/prometheus.groovy [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+						[$class: 'StringParameterValue', name: 'DURATION', value: DURATION ]]
+						} catch ( Exception e) {
```
Comment (chaitanyaenr) [inline alderaan/pipeline-scripts/prometheus.groovy]: Spacing issue.

File: alderaan/pipeline-scripts/prometheus.groovy (hunk 5 above)
Comment (mrsiano) [reply alderaan/pipeline-scripts/prometheus.groovy]: sure, to who?

File: alderaan/pipeline-scripts/prometheus.groovy [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+				echo "${JOB} Job failed with the following error: "
+				echo "${e.getMessage()}"
+				echo "Sending an email"
+				mail(
+					to: 'aos-scalability@redhat.com',
```
[the remark this answers is not in the document]
Comment (mrsiano) [reply alderaan/pipeline-scripts/prometheus.groovy]: sure, to who?

[the remark this answers is not in the document]
Comment (chaitanyaenr) [reply alderaan/pipeline-scripts/prometheus.groovy]: sejug@redhat.com, nelluri@redhat.com.

File: alderaan/pipeline-scripts/prometheus.groovy [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+			println "JOB: '${JOB}'"
+			println "REFRESH_INTERVAL: '${REFRESH_INTERVAL}'"
+			println "CONCURRENCY: '${CONCURRENCY}'"
+			println "GRAPH_PERIOD: '${GRAPH_PERIOD}'"
+			println "DURATION: '${DURATION}'"
+			println "-------------------------------------------------"
+			println "-------------------------------------------------"
+
+			// copy the parameters file to jump host
+			sh "git clone https://${token}@${repo} ${WORKSPACE}/perf-dept && chmod 600 ${WORKSPACE}/perf-dept/ssh_keys/id_rsa_perf"
```
Comment (chaitanyaenr) [inline alderaan/pipeline-scripts/prometheus.groovy]: s/repo/PERF_REPO
s/token/GITHUB_TOKEN

File: alderaan/pipeline-scripts/prometheus.groovy [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+			sh "scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${WORKSPACE}/perf-dept/ssh_keys/id_rsa_perf ${property_file_name} root@${jump_host}:/root/properties"
```
Comment (chaitanyaenr) [inline alderaan/pipeline-scripts/prometheus.groovy]: s/jump_host/JUMP_HOST

**Comment by chaitanyaenr** [comment]
@jeremyeder PTAL.

**Comment by jeremyeder** [comment]
thanks @mrsiano

**Pull request merged**